### PR TITLE
Add notes-rgx option for fixme checker

### LINKED ISSUE #2874: support more sophisticated fixme warnings
### Is your feature request related to a problem? Please describe
pylint warns for configured fixme notes (defaulting to FIXME/XXX/TODO) but sometimes it would be desirable to be more flexible in how these are flagged. For example, `# FIXME: this is broken` by itself might not be ok but `# FIXME: this is broken (ISSUE-1234)` might be acceptable because it points to an externally tracked issue

### Describe the solution you'd like
The `notes` config value should optionally allow a regex pattern to handle more complicated `fixme` conditions (eg `FIXME*ISSUE-\d+`).

### Additional context
The checker for this rule already [uses a regex pattern](https://github.com/PyCQA/pylint/blob/master/pylint/checkers/misc.py#L108-L112) to flag configured notes so I think this should be easy to implement, but patterns are currently escaped before being inserted into the regex.

### PR DESCRIPTION
**Pull request by BennyTheSen**
## Steps

- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description

This PR adds the option ``notes-rgx`` which can be used to flag the fixme warnings more flexible with a regex. 
Also extends the fixme functional test to include the ``notes`` and the ``notes-rgx`` option

## Type of Changes

|   | Type |
| ------------- | ------------- |
|   | :bug: Bug fix  |
| ✓  | :sparkles: New feature |
|   | :hammer: Refactoring  |
|   | :scroll: Docs |

### CODE REVIEW

File: doc/whatsnew/2.5.rst [hunk 1]
```diff
@@ -82,3 +82,5 @@ separated list of regexes, that if a name matches will be always marked as a bla
 * Add new --fail-under flag for setting the threshold for the score to fail overall tests. If the score is over the fail-under threshold, pylint will complete SystemExit with value 0 to indicate no errors.
 
 * Add a new check (non-str-assignment-to-dunder-name) to ensure that only strings are assigned to ``__name__`` attributes
+
+* Add a new option ``notes-rgx`` to make fixme warnings more flexible
```
Comment (PCManticore) [inline doc/whatsnew/2.5.rst]: Let's mention here that either `notes-rgx` or `notes` can be used.

File: pylint/checkers/misc.py [hunk 2]
```diff
@@ -94,12 +94,23 @@ class EncodingChecker(BaseChecker):
                 ),
             },
         ),
+        (
+            "notes-rgx",
+            {
+                "type": "string",
+                "metavar": "<regexp>",
+                "default": ("a^"),
```
Comment (PCManticore) [inline pylint/checkers/misc.py]: This should not have a default. Let's remove it from line 102.

File: pylintrc [hunk 3]
```diff
@@ -103,6 +103,8 @@ logging-modules=logging
 # List of note tags to take in consideration, separated by a comma.
 notes=FIXME,XXX,TODO
 
+# Regular expression of note tags to take in consideration.
+notes-rgx=a^
```
Comment (PCManticore) [inline pylintrc]: This needs to be removed.

File: tests/functional/f/fixme.py [hunk 4]
```diff
@@ -18,6 +18,15 @@ def function():
     #FIXME: no space after hash
     # +1: [fixme]
     #todo: no space after hash
+
+    # +1: [fixme]
```
Comment (PCManticore) [inline tests/functional/f/fixme.py]: These are some nice tests!

**Comment by BennyTheSen** [comment]
Hey @PCManticore thanks for the input, I resolved your comments.

**Comment by PCManticore** [comment]
Thanks a lot @BennyTheSen !

**Pull request merged**